library/src/main/kotlin/com/strumenta/kolasu/languageserver/KolasuServer.kt

```diff
@@ -259,13 +264,21 @@ open class KolasuServer<T : Node>(
         uri: String,
         text: String
     ) {
-        if (!::indexWriter.isInitialized) return
 
         val parsingResult = parser?.parse(text) ?: return
         files[uri] = parsingResult
 
         val tree = parsingResult.root ?: return
 
+        updateIndex(uri, tree)
+        reportDiagnostics(parsingResult, tree, uri)
+    }
+
+    private fun updateIndex(uri: String, tree: Node) {
+        if (!::indexWriter.isInitialized) {
```

Comment:
I did not know about this syntax, nice

Reply:
I merely moved (refactored) this code but I wasn't aware of that syntax either, I think it's code Martin wrote